feat: add work expression view with TOC navigation and preview

hamo225 opened the pull request:
RISDEV-8517

hamo225 committed:
feat: add work expression view with TOC navigation and preview

RISDEV-8517

hamo225 committed:
feat: add NormExpressionEli to DokumentExpressionEli helper

- also adds adjustment for min size of panels

RISDEV-8517

hamo225 committed:
refactor(datenbank): migrate WorkDetail to use child routing

- use RouterView for expression details
- Add child route DatenbankWorkExpressionDetail

RISDEV-8517

hamo225 committed:
Adds tests for service and scrollto

RISDEV-8517

malte-laukoetter commented on frontend/src/router.ts:
```diff
@@ -114,6 +114,19 @@ const routes: readonly RouteRecordRaw[] = [
+        component: () => import("@/views/datenbank/WorkDetail.view.vue"),
+        children: [
+          {
+            path: `expression/${createDokumentExpressionEliPathParameter("expression")}`,
```
one thought i just had: In theory it should be enough to just include the additional parts of the expression eli here (pointInTime, version & language) and use the existing parts of the work eli. That way the path for a work would look like `/datenbank/eli/bund/bgbl-1/2025/1` and for an expression of that work like `/datenbank/eli/bund/bgbl-1/2025/1/2025-03-12/1/deu`. That way we wouldn't need to include the work parts twice

hamo225 replied:
I do understand you. Lets pair on this on Wednesday. Or you can go ahead an make the change tomorrow if you have time. I will leave it as it is for now as i will not be here tomorrow.

malte-laukoetter replied:
I merged this pr and sure we can have a look into that together tomorrow :-)

malte-laukoetter commented on frontend/src/views/datenbank/WorkDetail.view.vue:
```diff
@@ -0,0 +1,161 @@
+
+  if (route.name === "DatenbankWorkExpressionDetail") {
+    const expressionEli = useDokumentExpressionEliPathParameter("expression")
+    console.log(expressionEli)
```
```suggestion
```

malte-laukoetter commented on frontend/src/views/datenbank/WorkExpressionDetail.view.vue:
```diff
@@ -0,0 +1,153 @@
+</template>
+
+<style scoped>
+:deep(.selected) {
```
could you scope this to just affect the stuff in the preview. Especially with `:deep` it is otherwise quite likely to affect the styling of something unrelated

malte-laukoetter commented on frontend/src/views/datenbank/WorkDetail.view.vue:
```diff
@@ -0,0 +1,161 @@
+  background-color: var(--color-element-select-selected-background);
+  outline: 2px solid var(--color-element-select-selected-border);
+  outline-offset: 2px;
+}
```
i dont think this is used in this view

malte-laukoetter commented on frontend/src/views/datenbank/WorkDetail.view.vue:
```diff
@@ -0,0 +1,161 @@
+function getInForceDateFromEli(eli: string) {
+  try {
+    return formatDate(NormExpressionEli.fromString(eli).pointInTime)
+  } catch {
```
when does this error case happen? And should we log something when it happens?

hamo225 replied:
when the ELI string that doesn't match the expected format, malformed, missing parts but i have now inlined the function within the computed treeNodesExpressions.

malte-laukoetter reviewed (commented):
Overall nice work! Just left some smaller comments and an idea i just had :)

hamo225 committed:
fix: address PR review feedback

- Remove debug console.log statement
- Scope :deep(.selected) to preview component
- Remove unused CSS styles from WorkDetail view
- Inline getInForceDateFromEli function in treeNodesExpressions computed

RISDEV-8517

hamo225 committed:
feat(e2e): add work details view e2e tests

RISDEV-8517
